text/0015-market-design-revisit.md

```diff
@@ -0,0 +1,116 @@
+
+Ultimately, this the framework proposed here adheres to all requirements stated in RFC-1.
+
+## Stakeholders
+
+Primary stakeholder sets are:
+
+- Protocol researchers and developers, largely represented by the Polkadot Fellowship and Parity Technologies' Engineering division.
+- Polkadot Parachain teams both present and future, and their users.
+- Polkadot DOT token holders.
+
+## Explanation
+
+### Bulk Markets
+ 
+The `BULK_PERIOD` has been restructured into two primary segments: the `MARKET_PERIOD` and `RENEWAL_PERIOD`, along with an auxiliary `SETTLEMENT_PERIOD`. This latter period doesn't necessitate any actions from the coretime system chain, but it facilitates a more efficient allocation of coretime in secondary markets. A significant departure from the original proposal lies in the timing of renewals, which now occur post-market phase. This adjustment aims to harmonize renewal prices with their market counterparts, ensuring a more consistent and equitable pricing model.
+
+#### Market Period (14 days)
+
+During the market period, core sales are conducted through a well-established **clearing price Dutch auction** that features a `RESERVE_PRICE`. The price initiates at a premium, designated as `PRICE_PREMIUM` (for instance, 30%) and descends linearly to the `RESERVE_PRICE` throughout the duration of the `MARKET_PERIOD`. Each bidder is expected to submit both their desired price and the quantity (that is, the amount of Coretime) they wish to purchase. To secure these acquisitions, bidders must make a deposit equivalent to their bid multiplied by the chosen quantity, in DOT. 
+
+The market achieves resolution once all quantities have been sold, or the `RESERVE_PRICE` has been reached. This situation leads to determining the `MARKET_PRICE` either by the lowest bid that was successful in clearing the entire market or by the `RESERVE_PRICE`. This mechanism yields a uniform price, shaped by market forces (refer to the following discussion for an explanation of its benefits). In other words, all buyers pay the same price (per unit of Coretime). Further down the benefits of this variant of a Dutch auction is discussed.
+
+**Note:** In cases where some cores remain unsold in the market, all buyers are obligated to pay the `RESERVE_PRICE`.
+
+#### Renewal Period (7 days)
+
+As the `RENEWAL_PERIOD` commences, all current tenants are granted the opportunity to renew their cores at a slight discount of `MARKET_PRICE * RENEWAL_DISCOUNT` (for instance, 10%). This provision affords marginal benefits to existing tenants, balancing out the non-transferability aspect of renewals.
```

Comment:
I think this is against our original goals. What we want (as far as I understand):

- Maximize core utilization
- Price predictability for projects, because people need to be able to plan.

The automatic renewal based on a continuous lease (no selling of slices) is already against goal 1, as it incentivizes people to not sell block space they actually don't need and will instead just produce empty blocks. This is already not great, but giving them a discount makes it even worse as it then is actually cheap to do that!

So having a guaranteed lease should be more expensive than ordering on the free market - at least on average. We would like for projects to use the free market as much as possible, as this maximizes utilization.

What we want though, is that long term projects don't suddenly have 10x fees, just because there was a spike in demand. It is questionable how big such spikes could even be on that time scale, but let's assume it is an issue. What I would be proposing is that renewals cost something like the average price of the last year's worth of sales plus some x%. This should give projects the required planning certainty, while still making it cheaper (on average) to just buy exactly what you need (and nothing more) on the market. 

We then also have the above x as a screw to adjust utilization. If governance considers it too low, we just make x larger.

Reply:
Thanks for your comments, let me address your comments with regard to your stated goals:

1) **Maximizing core utilization:** I agree that this sounds like an important goal. But, in my opinion, it is more nuanced than this. While Polkadot UC offers more supply of coretime than demand, this could be true. But in times of higher demand than supply, we need to add some important part to this goal: We want to achieve maximally efficient coreusage. That requires us to put all consumers under constant pressure to pay (at least roughly) the current market price of coretime. My proposed model tries to achieve that. With regard to `RENEWAL_DISCOUNT`, I was also hesitant to add it. We could set it to 0, of course. My rationale to give some discount was that (correct me if I am wrong), that renewed coretime cannot be sold on the secondary market. That makes them less fungible (and less flexible), therefore allowing for some discount. In addition, it helps to give current tenants more safety, something that I inferred was also part of the goals.
2) **Price predictability:** This goal is obviously contradictory to a fully functioning market and points raised before. My proposed mechanism, however, still gives some price predictability. The price in a year for a current tenant can be calculated (as higher bound) as follows: Assuming 100% core usage and the resulting increase in price of `RESERVE_PRICE` every month (the function is still left undetermined, both in my RFC and RFC-1) and a maximum premium of 30% at the end of the last month (minus a potential `RENEWAL_DISCOUNT`). In my opinion, this still leaves enough certainty, while allowing for more market forces. That means, current tenants are protected from "suddenly paying 10x fees", because the price within a `BULK_PERIOD` is capped by the `PRICE_PREMIUM` (say 30%). That means, they are *protected from suddenly paying* 10x the fees. But they might be over a longer period of time pay 10x the fees, but that means the market and demand for coretime is so high, that they **should be paying 10x the fees**. Because otherwise very good waiting projects cannot compete and get in.

Reply:
I somehow missed the boundedness. In this case we only need to make sure it is reasonably bounded, in order to be able to reasonably plan with the worst case.

For the discount part: I think you should be paying more: In exchange you know exactly (or at least way better) in advance what you are going to pay. Projects then just need to pick what they need: Basically paying an insurance fee against price surges or going cheap and take the risk. You can evaluate your decision based on your requirements from time to time, but there should be no need to re-adjust each month (as there was some concern with a previous proposal).

Ideally we would make it so that we can do away with the no selling of parts of the coretime, as this would benefit utilization, but only let people pay for price stability. In the end, I think it should be some kind of insurance: Where you always pay a premium, but in exchange your price curve is flattened/evened out. This could actually be a service on the secondary market already, but I think we consider this essential enough to put it into the core directly.

Reply:
thanks for the hint, I'll highlight in the RFC directly that the price is reasonably bounded (and mostly driven by the price adjustment of the reserve price that is identical to RFC-1).

With regard to the discount part:
1) I disagree to have current tenants pay more simply because of the fact that they would rather buy the core (maybe anonymously) in the `MARKET_PERIOD`, then not renew, and then jump into the new core. This creates unnecessary overhead for Polkdaot UC. Of course you could argue that they face some uncertainty of not getting the core they bought and are left without any, but its just messy. My goal was to create a system where the best strategy of a current tenant is to simply take the offer they get and not needing to participate in the auction again.
2) I tend to agree that we could remove the discount if we make the renewed coretime fully fungible. But I was under the impression that this is somewhat a technical restriction, and given this assumption together with the fact that we like to give some advantage to current tenants, giving a small discount seems reasonable. The "priority right to stay" when you pay exactly the market price is just not that much of a priority.

Reply:
I think what matters is that you can be certain to renew your slot. E.g. you not suddenly find yourself outbid and then you don't have a core for the next month. The other thing is just the already mentioned ability to plan ahead .. have some price certainty. I don't think a small discount on the most recent market price does anything to either goal.